crates/walrus-sdk/src/client/store_args.rs

```diff
@@ -72,6 +76,11 @@ pub struct StoreArgs {
     pub tail_handle_collector: Option<Arc<Mutex<Vec<JoinHandle<()>>>>>,
     /// Optional channel to forward encoding progress events to.
     pub encoding_event_tx: Option<UnboundedSender<EncodingProgressEvent>>,
+    /// Initial upload weight accumulated before certification (e.g., from pending uploads).
+    pub initial_upload_weight: Option<std::collections::HashMap<walrus_core::BlobId, usize>>,
+    /// Nodes that failed during pending uploads, keyed by blob ID, to scope immediate retries.
+    pub initial_failed_nodes:
+        Option<std::collections::HashMap<walrus_core::BlobId, Vec<NodeIndex>>>,
```

Comment:
remark: I find that passing these two HashMap, which are the implementation detail of the client execution, in StoreArgs is quite confusing. Can we not place them here, but use a dedicated data structure in `reserve_and_store_encoded_blobs`? The lifetime of these HashMap should be within `reserve_and_store_encoded_blobs`, is that correct?

Reply:
Created a new struct called `PendingContext` to store thins information.